examples/text-classification/run_glue.py

```diff
@@ -413,6 +414,25 @@ def compute_metrics(p: EvalPrediction):
     if training_args.do_predict:
         logger.info("*** Test ***")
 
+        # Get the datasets: you can provide your own CSV/JSON test file (see below)
+        # when you use `do_predict` without specifying a GLUE benchmark task.
+
+        if data_args.task_name is None and data_args.test_file is not None:
+            extension = data_args.test_file.split(".")[-1]
+            assert extension in ["csv", "json"], "`test_file` should be a csv or a json file."
+            if data_args.test_file.endswith(".csv"):
+                # Loading a dataset from a local csv file
+                test_dataset = load_dataset("csv", data_files={"test": data_args.test_file})
+            else:
+                # Loading a dataset from a local json file
+                test_dataset = load_dataset("json", data_files={"test": data_args.test_file})
```

Comment:
Can we put those lines earlier, with the validation dataset? This way the map will be done with the other dataset. I think we can do something nice by creating `data_files={"train": data_args.train_file, "validation": data_args.validation_file}` and then adding the keys `test` if the test_file is passed.

Reply:
Thank you for your comment! I've reflected the review.
The nested `if` statements in the code have increased, but I think the readability may have improved in terms of "whether to use GLUE task or to use local files". What do you think?

Reply:
I also added the logger output to make sure that the local files a user wants to use are loaded correctly. If this is superfluous, please let me know and I will remove it.